test/metrics/prom-metrics-linter/custom_linter_rules.go

```diff
@@ -56,10 +56,116 @@ func CustomLinterRules(problems []promlint.Problem, mf *dto.MetricFamily, operat
 		}
 	}
 
-	// Sort the problems by metric name
-	sort.Slice(newProblems, func(i, j int) bool {
-		return newProblems[i].Metric < newProblems[j].Metric
-	})
-
 	return newProblems
 }
+
+func CustomRecordingRuleValidation(rr recordingRule) []promlint.Problem {
+	var problems []promlint.Problem
+
+	name := rr.Record
+	if name == "" {
+		return problems
+	}
+
+	// Require name structure: level:metric:operations
+	parts := strings.Split(name, ":")
+	if len(parts) != 3 || parts[0] == "" || parts[1] == "" || parts[2] == "" {
+		problems = append(problems, promlint.Problem{Metric: name, Text: "recording rule name must be level:metric:operations. if there is no obvious operations, use 'sum'"})
+		return problems
+	}
+
+	// Enforce metric segment prefix: it should start with operator_suboperator like metrics do
+	{
+		metricPart := parts[1]
+		expectedPrefix := operatorName + "_"
+		if operatorName != subOperatorName {
+			expectedPrefix = operatorName + "_" + subOperatorName + "_"
+		}
+		if !strings.HasPrefix(metricPart, expectedPrefix) {
+			problems = append(problems, promlint.Problem{Metric: name, Text: fmt.Sprintf("metric (second segment in level:metric:operations) need to start with \"%s\"", expectedPrefix)})
+		}
+	}
+
+	// Single/Multiple operations: detect ops in expr and enforce suffix rules
+	{
+		aggOps := detectOps(rr.Expr, promAggOps)
+		timeOps := detectOps(rr.Expr, promTimeOps)
+		total := len(aggOps) + len(timeOps)
+		opsPart := parts[2]
+		if total == 1 {
+			if len(aggOps) == 1 {
+				expected := aggOps[0]
+				if !strings.HasSuffix(opsPart, expected) {
+					problems = append(problems, promlint.Problem{Metric: name, Text: fmt.Sprintf("single aggregation detected in expr: operations (third segment in level:metric:operations) should end with '%s'", expected)})
+				}
+			} else if len(timeOps) == 1 {
+				expected := timeOps[0]
+				timeSuffixRe := regexp.MustCompile(`(?i)` + expected + `[0-9]+[smhdwy]+$`)
```

Comment:
Not sure about using MustCompile with dynamic regex. I think it would be better to use Compile + error check.